.rubocop.yml

```diff
@@ -0,0 +1,153 @@
+inherit_from: .rubocop_todo.yml
+
+AllCops:
+  Exclude:
+    # Exclude .gemspec files because they are generally auto-generated
+    - '*.gemspec'
+
+# Generally, the keyword style uses a lot of space. This is particularly true when
+# you use case/if statements, in combination with a long-name variable.
+#
+#   invoice_error_message = case error
+#                           when 1 == 1
+#                             do_something
+#                           else
+#                             do_else
+#                           end
+#
+Lint/EndAlignment:
+  AlignWith: variable
+
+# [codesmell]
+Metrics/AbcSize:
+  Enabled: false
+  Exclude:
+    - 'spec/**/*_spec.rb'
+    - 'test/**/*_test.rb'
+
+# [codesmell]
+Metrics/ClassLength:
+  Enabled: false
+  Exclude:
+    - 'spec/**/*_spec.rb'
+    - 'test/**/*_test.rb'
+
+# [codesmell]
+Metrics/MethodLength:
+  Enabled: false
+  Exclude:
+    - 'spec/**/*_spec.rb'
+    - 'test/**/*_test.rb'
+  Max: 15
```

Comment:
@jacegu I'm fine to decrease it, I believe the default is 10, whereas it was set to 25 (since we have one method that is 25 lines long).

I personally have a bad relationship with code metrics. They need to be interpreted, I would not extract a single 25-lines method in 3 methods (used only in that particular circumstance) only to please a metric that wants a method to not be longer than 10 lines. To me, it's like hiding the dust under the carpet.

That's also why I disabled it by default, and flagged it as "codesmell". Codesmell metrics are not enabled by default, they should be manually enabled and run. I was also considering to decrease the level of codesmell cops from Convention to Refactor, and not run them by default on the builds.

I'd like to keep them as indicators, but I don't want them to cause a build to fail.

Do you feel we should decrease it?